grails-app/controllers/org/pih/warehouse/api/LocalizationApiController.groovy

```diff
@@ -21,30 +21,41 @@ class LocalizationApiController {
     GrailsApplication grailsApplication
 
     def list() {
-        String languageCode = params.lang
+        String languageCode = params.languageCode
         String prefix = params.prefix
 
         String[] supportedLocales = grailsApplication.config.openboxes.locale.supportedLocales
 
+        Locale defaultLocale = Locale.default
         Locale currentLocale = localizationService.getCurrentLocale()
-        Locale defaultLocale = currentLocale.getDefault()
-        Locale selectedLocale = languageCode ? LocalizationUtil.getLocale(languageCode) : currentLocale
+        Locale selectedLocale = languageCode ? StringUtils.parseLocaleString(languageCode) : currentLocale
 
-        // Get the default message properties as well as the message properties for the selected locale
+        // Get the system default message properties
         Properties defaultMessageProperties = localizationService.getMessagesProperties(defaultLocale)
+
+        // Get the fallback language default message properties if it is country specific language
+        Properties fallbackMessageProperties = null
+        if (selectedLocale.country && supportedLocales.contains(selectedLocale.language)) {
+            Locale fallbackLocale = new Locale(selectedLocale.language)
+            fallbackMessageProperties = localizationService.getMessagesProperties(fallbackLocale)
+        }
+        // Get the message properties for the selected locale
         Properties selectedMessageProperties = localizationService.getMessagesProperties(selectedLocale)
 
         // Get all translations for the given prefix and locale from the database
-        List<Localization> localizedMessages = prefix ? Localization.findAllByCodeIlikeAndLocale("${prefix}%", selectedLocale.language) :
-                Localization.findAllByLocale(selectedLocale.language)
+        List<Localization> localizedMessages = prefix ? Localization.findAllByCodeIlikeAndLocale("${prefix}%", selectedLocale.toString()) :
+                Localization.findAllByLocale(selectedLocale.toString())
```

Comment:
The same question as under the `activeLanguageTag` but in the opposite way

Reply:
This will resolve into `es_MX` and i think this is how it should look like in this case